lib/zendesk_apps_support/manifest/no_override_hash.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+require 'json'
+
+module ZendeskAppsSupport
+  class Manifest
+    class OverrideError < StandardError
+      class << self
+        def message_for(key, original = nil, attempted = nil)
+          message = "Duplicate reference in manifest: #{key}"
+          if original && attempted
+            message = "#{message}. Initially set to #{original}, attempted overwrite to #{attempted}."
+          end
+          message
+        end
+      end
+    end
+
+    class NoOverrideHash < Hash
+      class << self
+        def [](array)
+          uniques = Set.new
+          array.each do |key, _value|
+            if uniques.add?(key).nil?
+              # don't add the value to the error message because it might be '_legacy'
+              raise OverrideError, OverrideError.message_for(key)
+            end
+          end
+          super
```

Comment:
I love this class, but it seems a bit strange to implement the same check as `[]=` in a very different way. Could we build up our own result (instead of calling `super`) using `[]=` to do the check for us?

(it's a shame `super` doesn't already use `[]=` internally)

Not sure what to do about the desired different error message in that case though - we could catch and rethrow a less descriptive OverrideError, but to be honest that logic probably belongs in the user (manifest.rb) rather than this class anyway.